WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -86,105 +117,162 @@ private extension ReviewsDashboardCardViewModel {
-                                                                          pageNumber: 1,
-                                                                          pageSize: Constants.numberOfItems) { result in
-                switch result {
-                case .success:
-                    // Ignoring the result from remote as we're using storage as the single source of truth
-                    continuation.resume()
-                case .failure(let error):
-                    continuation.resume(throwing: error)
+    /// Populates data from the local storage for the three main data (reviews, products, notifications).
+    ///
+    func updateData() {
+        let localReviews = reviews.prefix(Constants.numberOfItems)
+
+        // We can show partial review content as long as there are reviews found in storage.
+        if localReviews.isEmpty == false {
+            showLoadingAnimation = false
+        }
+
+        data = localReviews.map { review in
+
+            // Depending on the sync progress, `product` and `notification` might still be nil.
+            // This is acceptable and the app is able to display partial review content.
+            let product = reviewProducts.first { $0.productID == review.productID }
+            let notification = notifications.first { notification in
+                if let notificationReviewID = notification.meta.identifier(forKey: .comment) {
+                    return notificationReviewID == review.reviewID
                 }
-            })
+                return false
+            }
+            return ReviewViewModel(
+                showProductTitle: product != nil,
+                review: review,
+                product: product,
+                notification: notification
+            )
         }
     }
 
-    /// Updates data
     @MainActor
-    func updateReviewsResults() async {
-        let reviews = productReviewsResultsController.fetchedObjects.prefix(Constants.numberOfItems)
-        let productIDs = reviews.map { $0.productID }
+    func synchronizeReviews(filter: ReviewsFilter) async throws {
+        try await synchronizeProductReviews(filter: filter)
+
+        let productIDs = reviews.prefix(Constants.numberOfItems).map { $0.productID }
 
         if productIDs.isNotEmpty {
-            updateProductsResultsControllerPredicate(with: productIDs)
-            do {
-                try await fetchProducts(for: productIDs)
-            } catch {
-                ServiceLocator.crashLogging.logError(error)
-            }
-        }
+            // As long as the app is able to fetch one review once, then the button should always appear.
+            // Later on, when filtering by hold or pending status, the reviews result might become empty.
+            // In that case, the app should keep showing the button to allow the user to see all non-filtered reviews.
+            shouldShowAllReviewsButton = true
```

Comment:
Similar comment as above about separation of concern. For simplicity, I'd suggest always keeping the button available.

Reply:
Ah yup, I just checked that other cards keep the button available even when nothing is available. Updated https://github.com/woocommerce/woocommerce-ios/pull/12769/commits/b61a44ae35b91ea7d9014effc6589aa511b3c495